lib/src/rules/avoid_types_as_parameter_names.dart

```diff
@@ -68,14 +68,11 @@ class _Visitor extends SimpleAstVisitor<void> {
 
   @override
   void visitFormalParameterList(FormalParameterList node) {
-    if (node.parent is GenericFunctionType) return;
-
     for (var parameter in node.parameters) {
       var declaredElement = parameter.declaredElement;
       var name = parameter.name;
       if (declaredElement != null &&
           declaredElement is! FieldFormalParameterElement &&
-          declaredElement.hasImplicitType &&
```

Comment:
Would you mind reverting this bit; it shouldn't be needed for the fix, and the motivation of the lint rule is about implicit typed parameters.

Reply:
Done, please check the changed test cases in the same commit.